src/tuesday/core/instrument_models/noise.py

```diff
@@ -0,0 +1,197 @@
+        Effective area of the antenna with shape (Nfreqs,).
+        If provided, we use
+        $\Omega_{\rm beam} = \lambda^2/A_{\rm eff}$.
+        If not provided, we use
+        $\Omega_{\rm beam} = 0.004 \cdot (\nu/150 MHz)^{-2} \cdot rad^2$,
+        which comes from assuming that the effective area is approximately 1000 m$^2$.
+    """
+    try:
+        len(freqs)
+    except TypeError:
+        freqs = np.array([freqs.value]) * freqs.unit
+
+    sig_uv = np.zeros(len(freqs))
+    for i, nu in enumerate(freqs):
+        obs = observation.clone(
+            observatory=observation.observatory.clone(
+                beam=observation.observatory.beam.clone(frequency=nu)
+            )
+        )
+        tsys = obs.Tsys.to(un.mK)
+        if a_eff is None:
+            f0 = 150.0 * un.MHz
+            # =\lambda^2/A_{eff}, this approx assumes A_{eff} ~ 1000 m^2
+            omega_beam = 0.004 * (nu.to(un.MHz) / f0) ** (-2.0) * un.rad**2
+        else:
+            wavelength = (c / nu.to("Hz")).to(un.m)
+            omega_beam = (wavelength**2 / a_eff[i].to(un.m**2)) * un.rad**2
+
+        d = Planck18.comoving_distance(f2z(nu)).to(un.Mpc)  # Mpc
+        theta_box = (boxlen.to(un.Mpc) / d) * un.rad
+        omega_pix = theta_box**2 / np.prod(lc_shape[:2])
+
+        sqrt = np.sqrt(2.0 * observation.bandwidth.to("Hz") * obs.integration_time).to(
+            un.dimensionless_unscaled
+        )
+        # I need this 1e6 to get the same numbers as tools...
+        sig_uv[i] = tsys.value * omega_beam / omega_pix / sqrt / 1e6
+    return sig_uv * tsys.unit
+
+
+def blackmanharris(n: int):
+    r"""Blackman-Harris window function for a 2D grid.
+
+    Parameters
+    ----------
+    n : int
+        Size of the window function, assumed to be square.
+
+    Returns
+    -------
+    wf : np.ndarray
+        2D Blackman-Harris window function with shape (n, n)
+
+    """
+    wf = np.abs(windows.blackmanharris(n))
+    return np.sqrt(np.outer(wf, wf))
+
+
+def sample_lc_noise(
+    rms_noise: un.Quantity,
+    seed: int | None = None,
+    nsamples: int = 1,
+    window_fnc: Callable | None = None,
+):
+    """Sample noise for a lightcone slice given the corresponding rms noise in uv space.
+
+    Parameters
+    ----------
+    rms_noise : astropy.units.Quantity
+        RMS noise in uv space, shape (Nx, Ny, Nfreqs).
+    nsamples : int, optional
+        Number of noise realisations to sample, by default 1.
+    window_fnc : Callable, optional
+        Window function to be applied to the noise sampled in uv space,
+        by default windows.blackmanharris.
+
+    Returns
+    -------
+    lc_noise : un.Quantity
+        Noise sampled in real space, shape (nsamples, Nx, Ny, Nfreqs
+
+    """
+    if len(rms_noise.shape) == 2:
+        rms_noise = rms_noise[..., None]
+    if seed is None:
+        seed = np.random.default_rng().integers(0, 2**31 - 1)
+        warnings.warn(f"Setting random seed to {seed}", stacklevel=2)
```

Comment:
This will end up being annoying. Maybe just make a logger.info() statement here.